contracts/coconsensus/Coconsensus.sol

```diff
@@ -46,23 +46,171 @@ contract Coconsensus is MasterCopyNonUpgradable, GenesisCoconsensus, MosaicVersi
     }
 
 
+    /* Constants */
+
+    /**
+     * Sentinel pointer for marking the ending of circular,
+     * linked-list of genesis metachain ids.
+     */
+    bytes32 public constant SENTINEL_METACHAIN_ID = bytes32(0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff);
+
+
     /* Storage */
 
+    /** Metachain id of the origin chain. */
+    bytes32 public originMetachainId;
```

Comment:
I believe we can go without explicitly having `originMetachainId` and `auxiliaryMetachainId` in coconsensus.

Reply:
We need to have this reference if we access blockchains per mcid. Also to check msg.sender in modifiers.

I would opt for originMetachainId and selfMetachainId though

Reply:
Ok keeping it `originMetachainId` and `selfMetachainId`.

Reply:
I see a reason to have selfMetachainId, however, not seeing the reason of having originMetachainId. However, let's keep them and see.